packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2911,6 +2913,40 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return null;
   }
 
+  TextRange _getParagraphBoundary(TextPosition position) {
+    return TextRange(start: _getParagraphStartOffset(position), end: _getParagraphEndOffset(position));
+  }
+
+  int _getParagraphStartOffset(TextPosition position) {
+    int offset = position.offset;
+    int? codeUnit = text!.codeUnitAt(offset);
```

Comment:
`text` is of type `TextSpan` so the returned `codeUnit` might be null. When it returns null, I think it means we have passed the first/last code unit. I wasn't sure if there's a good way to iterate over these since `TextSpan` didn't have a length property.

Reply:
You're right I was looking at the wrong method.